tests/repository/test_upload.py

```diff
@@ -130,3 +131,24 @@ def test_get_relative_url_with_file_object(tmpdir):
         repo._get_relative_url(file_obj, "some/path/")
 
     assert repo._get_relative_url(file_obj, "path/foo.txt") == "path/foo.txt"
+
+
+def test_upload_file_contains_unicode(client, requests_mocker):
+    file_obj = StringIO(u"some unicode")
```

Comment:
Ideally you should put some real non-ASCII text in here, because u"some unicode" can be interpreted without problems by code which can only handle ASCII or latin1 - so this test doesn't really ensure that unicode data is handled correctly.

Test would be a little stronger if you threw in an emoticon, some Chinese or whatever.